virt-launcher requires NET_RAW and assumes it is default enabled

<!-- This form is for bug reports and feature requests ONLY!
Also make sure that you visit our User Guide at https://kubevirt.io/user-guide/
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug


**What happened**:

Our DHCP service uses a raw socket and implements a UDP DHCP server in userspace, and it will fail when the cluster is configured to disable NET_RAW by default.

Got this error from virt-launcher on one such cluster:

panic: Unable to create new UDP4 listener: Unable to set socket option BINDTODEVICE: operation not permitted

**What you expected to happen**:

DHCP server starts and VM is created

**How to reproduce it (as minimally and precisely as possible)**:

Disable the NET_RAW capability in the container runtime, try to run a VM

--- a/pkg/virt-controller/services/template.go
+++ b/pkg/virt-controller/services/template.go
@@ -60,6 +60,7 @@ const debugLogs = "debugLogs"
 const MultusNetworksAnnotation = "k8s.v1.cni.cncf.io/networks"
 
 const CAP_NET_ADMIN = "NET_ADMIN"
+const CAP_NET_RAW = "NET_RAW"
 const CAP_SYS_NICE = "SYS_NICE"
 
 // LibvirtStartupDelay is added to custom liveness and readiness probes initial delay value.
@@ -1084,6 +1085,10 @@ func getRequiredCapabilities(vmi *v1.VirtualMachineInstance) []k8sv1.Capability 
 		(vmi.Spec.Domain.Devices.AutoattachPodInterface == nil) ||
 		(*vmi.Spec.Domain.Devices.AutoattachPodInterface == true) {
 		res = append(res, CAP_NET_ADMIN)
+		// The DHCP server needs the ability to use raw sockets. This
+		// capability is available by default in some clusters, but not
+		// a given.
+		res = append(res, CAP_NET_RAW)
 	}
 	// add a CAP_SYS_NICE capability to allow setting cpu affinity
 	res = append(res, CAP_SYS_NICE)

--- a/pkg/virt-controller/services/template_test.go
+++ b/pkg/virt-controller/services/template_test.go
@@ -1517,14 +1517,10 @@ var _ = Describe("Template", func() {
 				Expect(ok).To(BeTrue())
 				Expect(int(tun.Value())).To(Equal(1))
 
-				found := false
 				caps := pod.Spec.Containers[0].SecurityContext.Capabilities
-				for _, cap := range caps.Add {
-					if cap == CAP_NET_ADMIN {
-						found = true
-					}
-				}
-				Expect(found).To(BeTrue(), "Expected compute container to be granted NET_ADMIN capability")
+
+				Expect(caps.Add).To(ContainElement(kubev1.Capability(CAP_NET_ADMIN)), "Expected compute container to be granted NET_ADMIN capability")
+				Expect(caps.Add).To(ContainElement(kubev1.Capability(CAP_NET_RAW)), "Expected compute container to be granted NET_RAW capability")
 			})
 
 			It("Should require tun device if explicitly requested", func() {
@@ -1545,14 +1541,10 @@ var _ = Describe("Template", func() {
 				Expect(ok).To(BeTrue())
 				Expect(int(tun.Value())).To(Equal(1))
 
-				found := false
 				caps := pod.Spec.Containers[0].SecurityContext.Capabilities
-				for _, cap := range caps.Add {
-					if cap == CAP_NET_ADMIN {
-						found = true
-					}
-				}
-				Expect(found).To(BeTrue(), "Expected compute container to be granted NET_ADMIN capability")
+
+				Expect(caps.Add).To(ContainElement(kubev1.Capability(CAP_NET_ADMIN)), "Expected compute container to be granted NET_ADMIN capability")
+				Expect(caps.Add).To(ContainElement(kubev1.Capability(CAP_NET_RAW)), "Expected compute container to be granted NET_RAW capability")
 			})
 
 			It("Should not require tun device if explicitly rejected", func() {
@@ -1572,14 +1564,10 @@ var _ = Describe("Template", func() {
 				_, ok := pod.Spec.Containers[0].Resources.Limits[TunDevice]
 				Expect(ok).To(BeFalse())
 
-				found := false
 				caps := pod.Spec.Containers[0].SecurityContext.Capabilities
-				for _, cap := range caps.Add {
-					if cap == CAP_NET_ADMIN {
-						found = true
-					}
-				}
-				Expect(found).To(BeFalse(), "Expected compute container to not be granted NET_ADMIN capability")
+
+				Expect(caps.Add).To(Not(ContainElement(kubev1.Capability(CAP_NET_ADMIN))), "Expected compute container not to be granted NET_ADMIN capability")
+				Expect(caps.Add).To(Not(ContainElement(kubev1.Capability(CAP_NET_RAW))), "Expected compute container not to be granted NET_RAW capability")
 			})
 		})
 

--- a/pkg/virt-operator/creation/components/scc.go
+++ b/pkg/virt-operator/creation/components/scc.go
@@ -74,7 +74,7 @@ func NewKubeVirtControllerSCC(namespace string) *secv1.SecurityContextConstraint
 	scc.SELinuxContext = secv1.SELinuxContextStrategyOptions{
 		Type: secv1.SELinuxStrategyRunAsAny,
 	}
-	scc.AllowedCapabilities = []corev1.Capability{"NET_ADMIN", "SYS_NICE"}
+	scc.AllowedCapabilities = []corev1.Capability{"NET_ADMIN", "NET_RAW", "SYS_NICE"}
 	scc.AllowHostDirVolumePlugin = true
 	scc.Users = []string{fmt.Sprintf("system:serviceaccount:%s:kubevirt-controller", namespace)}
 

--- a/tests/vmi_networking_test.go
+++ b/tests/vmi_networking_test.go
@@ -553,14 +553,10 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 					_, ok = container.Resources.Limits[services.TunDevice]
 					Expect(ok).To(BeFalse())
 
-					netAdminCap := false
 					caps := container.SecurityContext.Capabilities
-					for _, cap := range caps.Add {
-						if cap == "NET_ADMIN" {
-							netAdminCap = true
-						}
-					}
-					Expect(netAdminCap).To(BeFalse(), "Compute container should not have NET_ADMIN capability")
+
+					Expect(caps).To(Not(ContainElement(k8sv1.Capability("NET_ADMIN"))), "Compute container should not have NET_ADMIN capability")
+					Expect(caps).To(Not(ContainElement(k8sv1.Capability("NET_RAW"))), "Compute container should not have NET_RAW capability")
 				}
 			}
 

--- a/tests/vmi_networking_test.go
+++ b/tests/vmi_networking_test.go
@@ -555,8 +555,8 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 
 					caps := container.SecurityContext.Capabilities
 
-					Expect(caps).To(Not(ContainElement(k8sv1.Capability("NET_ADMIN"))), "Compute container should not have NET_ADMIN capability")
-					Expect(caps).To(Not(ContainElement(k8sv1.Capability("NET_RAW"))), "Compute container should not have NET_RAW capability")
+					Expect(caps.Add).To(Not(ContainElement(k8sv1.Capability("NET_ADMIN"))), "Compute container should not have NET_ADMIN capability")
+					Expect(caps.Add).To(Not(ContainElement(k8sv1.Capability("NET_RAW"))), "Compute container should not have NET_RAW capability")
 				}
 			}
 

